### PR TITLE
[2404] Update Linux Installation Instructions

### LINKED ISSUE #2404: linux installation instructions are self-contradicting
* [ ] https://doc.owncloud.com/server/admin_manual/installation/deployment_recommendations.html#general-recommendations recommends has three scenarios only recommending RHEL and SLES,
**but** we give details only for Ubuntu 18.04 in https://doc.owncloud.com/server/admin_manual/installation/manual_installation.html#install-the-required-packages and have a dedicated examples setion also only on https://doc.owncloud.com/server/admin_manual/installation/ubuntu_18_04.html

* [x] https://doc.owncloud.com/server/admin_manual/installation/manual_installation.html#install-the-required-packages mentions "many people use automatic updates",
**but** https://doc.owncloud.com/server/admin_manual/installation/linux_installation.html#avoid-automatic-upgrades strongly suggests to avoid automatic updates.

* [x] https://doc.owncloud.com/server/admin_manual/installation/ubuntu_18_04.html claims to be "an ultra-short guide",
**but** with a total of 28 commands and 4 script files it is obviously quite comrehensive.

### PR DESCRIPTION
**Pull request by settermjd**
This PR fixes #2404. (_So far,_) it:

- Adds the instructions from @jnweiger as a truly ultra-short installation guide to the Ubuntu guide
- Refactors the installation guide to support multiple distributions and distribution versions
- Removes the phrase "_many people use automatic updates_" from https://github.com/owncloud/docs/blob/master/modules/admin_manual/pages/installation/manual_installation.adoc

When complete, it will have a similar guide for Red Hat / CentOS / Fedora, and SLES.

### CODE REVIEW

File: modules/admin_manual/pages/installation/ubuntu_18_04.adoc [hunk 1]
```diff
@@ -73,7 +104,7 @@ apt install -y \
   php-smbclient coreutils php-ldap
 ----
 
-WARNING: Ubuntu 18.04 includes smbclient 4.7.6, which has a known limitation of only using version 1 of the SMB protocol.
+WARNING: Ubuntu {ubuntu-version} includes smbclient 4.7.6, which has a known limitation of only using version 1 of the SMB protocol.
```
Comment (voroyam) [inline modules/admin_manual/pages/installation/ubuntu_18_04.adoc]: I disagree with replacing 18.04 with a variable here. ASFAIK this limitation is only in 18.04

**Comment by voroyam** [comment]
@settermjd are you sure that replacing the Ubuntu version with variables would work for your included files?

File: modules/admin_manual/pages/installation/ubuntu_18_04.adoc (hunk 1 above)
Comment (settermjd) [reply modules/admin_manual/pages/installation/ubuntu_18_04.adoc]: Fair enough.

Comment (jnweiger) [reply modules/admin_manual/pages/installation/ubuntu_18_04.adoc]: I a agree with voroyam to disagree. smbclient version number is specific to 18.04, document name is specific to 18.04

**Comment by voroyam** [comment]
Any reason why this document was put in a "guides" folder? What other guides are planned to go in to this folder?

**Comment by settermjd** [comment]
> Any reason why this document was put in a "guides" folder? What other guides are planned to go in to this folder?

Because of the following from the PR description:

> When complete, it will have a similar guide for Red Hat / CentOS / Fedora, and SLES.

File: modules/admin_manual/pages/installation/ubuntu_18_04.adoc (hunk 1 above)
Comment (settermjd) [reply modules/admin_manual/pages/installation/ubuntu_18_04.adoc]: In the most recent commit, https://github.com/owncloud/docs/pull/2480#discussion_r394988085 has been addressed.

**Comment by settermjd** [comment]
> @settermjd are you sure that replacing the Ubuntu version with variables would work for your included files?

Absolutely. Have a look at 0fc2bec to see how it builds on that change.

**Comment by settermjd** [comment]
> The content looks good.
> There is just @voroyam comment to address - what to do about the very specific warning about smbclient 4.7.6. We don't want someone to bump `ubuntu_version` and leave that warning, which will be out-of-date.

Well said. Have a look at 0fc2bec to see how that has been addressed.

**Comment by settermjd** [comment]
The intent of the most recent change is to be able to reuse one core documentation file in as many contexts as required, while also making it flexible and adaptable for the future.

**Comment by settermjd** [comment]
Thanks, @phil-davis, not finished yet, though. I'm adding in instructions for RHEL7 (using CentOS) and SLES 12.

**Comment by phil-davis** [comment]
> Thanks, @phil-davis, not finished yet, though. I'm adding in instructions for RHEL7 (using CentOS) and SLES 12.

This will be the all-singing-all-dancing PR with Mars rocket launch instructions also ;)

**Comment by settermjd** [comment]
> > Thanks, @phil-davis, not finished yet, though. I'm adding in instructions for RHEL7 (using CentOS) and SLES 12.
> 
> This will be the all-singing-all-dancing PR with Mars rocket launch instructions also ;)

I'm actually trying to do as little as possible, tbh. I'm doing my best to use the right aspects of the AsciiDoc format to keep it as simple as possible, yet, wrt maintenance, also be flexible, given what I've learned over the last few years.

**Comment by settermjd** [comment]
I've just about finished the RHEL section, based on CentOS 7 and will have the SLES section finished tomorrow. That should finalise the PR.

**Comment by phil-davis** [comment]
@settermjd request me for review again when the PR is ready.

**Comment by voroyam** [comment]
> When complete, it will have a similar guide for Red Hat / CentOS / Fedora, and SLES.

And why do you want to do those guides? The official docs are Ubuntu only.

**Pull request closed without merging**